msvc/vsimporter-templates/WinStore10-App/App.xaml.cpp

```diff
@@ -25,32 +25,28 @@ using namespace Windows::UI::Xaml::Navigation;
 /// Initializes the singleton application object.  This is the first line of authored code
 /// executed, and as such is the logical equivalent of main() or WinMain().
 /// </summary>
-App::App()
-{
+App::App() {
     InitializeComponent();
     Suspending += ref new SuspendingEventHandler(this, &App::OnSuspending);
 }
 
 extern "C" int main(int argc, char* argv[]);
 extern "C" int __cdecl EbrDefaultXamlMain();
 extern "C" void EbrApplicationActivated(Windows::ApplicationModel::Activation::IActivatedEventArgs^ args);
-extern "C" bool EbrApplicationLaunched(Windows::ApplicationModel::Activation::LaunchActivatedEventArgs^ args);
+extern "C" void EbrApplicationLaunched(Windows::ApplicationModel::Activation::LaunchActivatedEventArgs^ args);
 
 /// <summary>
 /// Invoked when the application is launched normally by the end user.  Other entry points
 /// will be used such as when the application is launched to open a specific file.
 /// </summary>
 /// <param name="e">Details about the launch request and process.</param>
-void App::OnLaunched(Windows::ApplicationModel::Activation::LaunchActivatedEventArgs^ e)
-{
-    if (EbrApplicationLaunched(e)) {
-        //  Jump default "C" main
-        main(0, NULL);
-    }
+void App::OnLaunched(Windows::ApplicationModel::Activation::LaunchActivatedEventArgs^ e) {
+    main(0, NULL);
+    EbrApplicationLaunched(e);
```

Comment:
Yes.  It was just checking if we were already launched and main did the actual launching, but I changed it to be consistent with EbrApplicationActivated and launch from there, and main does the setup

---
In reply to: [72324753](https://github.com/Microsoft/WinObjC/pull/686#discussion_r72324753) [](ancestors = 72324753)